app/Http/Controllers/TradeController.php

```diff
@@ -15,7 +16,28 @@ class TradeController extends Controller
      */
     public function index(Request $request)
     {
-        $trades = Trade::all();
+        $trades = Trade::where('status', 'open')
+                    ->has('items')->with('items')
```

Comment:
should be loading the item relation instead, and since the item relation will be essentially required, we wont need to check its existence
```suggestion
                    ->with('item')
```